packages/neutron-sdk/src/interchain_queries/v047/types.rs

```diff
@@ -0,0 +1,133 @@
+use std::ops::Div;
+// import all types from v045 package
+// to make it available from v047 package (kinda proxy) since they work with Cosmos SDK 0.47 as usual
+pub use crate::interchain_queries::v045::types::*;
+
+// But at the same time we replace some structs from v045 with structs below to create structures
+// compatible with Cosmos SDK 0.47
+
+use crate::interchain_queries::types::KVReconstruct;
+use crate::{bindings::types::StorageValue, errors::error::NeutronResult, NeutronError};
+
+use crate::interchain_queries::helpers::uint256_to_u128;
+use crate::interchain_queries::v047::helpers::deconstruct_account_denom_balance_key;
+use cosmos_sdk_proto::cosmos::staking::v1beta1::{
+    Delegation, Params, Validator as CosmosValidator,
+};
+use cosmos_sdk_proto::traits::Message;
+use cosmwasm_std::{Addr, Coin, Decimal256, Uint128, Uint256};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use std::str::FromStr;
+
+/// Key for Staking Params in the **staking** module's storage
+/// <https://github.com/cosmos/cosmos-sdk/blob/54120626e9994b2f1cc7a2bebc60cfb99703028f/x/staking/types/keys.go#L56>
+pub const STAKING_PARAMS_KEY: u8 = 0x51;
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+/// A structure that can be reconstructed from **StorageValues**'s for the **Balance Interchain Query**.
+/// Contains amounts of coins that are held by some account on remote chain.
+pub struct Balances {
+    pub coins: Vec<Coin>,
+}
+
+impl KVReconstruct for Balances {
+    fn reconstruct(storage_values: &[StorageValue]) -> NeutronResult<Balances> {
+        let mut coins: Vec<Coin> = Vec::with_capacity(storage_values.len());
+
+        for kv in storage_values {
+            let (_, denom) = deconstruct_account_denom_balance_key(kv.key.to_vec())?;
```

Comment:
kv.value can be empty which will lead to an error
```
fn tt() {
    let s = Binary::from([]);
    let x = Uint128::from_str(&String::from_utf8(s.to_vec()).unwrap()).unwrap();
}
```

Reply:
In what cases it can be empty?

Reply:
when you request balance of some denom which doesn't exist on remote chain address

Reply:
well, that's an error indeed?
Or what do you suggest here?

Reply:
no, it's not. this way if you are subscribed for some balance and it's gone on remote chain, then it won't be updated on neutron (as per parsing error)

Reply:
add condition 
if kv.value.len() > 0 {
  ... parsing here
}

Reply:
related https://github.com/neutron-org/neutron-sdk/pull/130/files#diff-a581f2e1fcde88fec1f98541f7d3a0e0c1703a173f47e94bd38ab9f57d014042R119

Reply:
Ok, i see, thanks. I'll fix it

Reply:
But we already have a testcase for an empty value: https://github.com/neutron-org/neutron-sdk/pull/133/files#diff-bed6220b3de5c7dafcf38be35cecaea9d4f395226d9a64175bfcfb517dd86dafR70

And it works fine